src/key_packages/errors.rs

```diff
@@ -0,0 +1,92 @@
+//! # Key Package errors
+//!
+//! `KeyPackageError` are thrown on errors handling `KeyPackage`s and
+//! `KeyPackageBundle`s.
+
+use crate::{config::ConfigError, extensions::ExtensionError};
+use std::error::Error;
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+#[repr(u16)]
+pub enum KeyPackageError {
+    /// An unknown extension error occurred.
+    UnknownExtensionError = 0,
```

Comment:
What is the purpose of this error? It doesn't seem to be used anywhere.

Reply:
It is used when the `ExtensionError` is not an `InvalidExtensionType`. It should never happen, but theoretically can because the `match` in the respective from has to be exhaustive.